### DISCUSSION TITLE
Window Position , Size , Callback

### DISCUSSION THREAD

**Discussion by aliessmael** [dotnet/maui#8723] [Q&A] [answered]
Hi , I am making PC app with multi window, 
I want to adjust positions, sizes on startup, also I want callback to check if these properties are changed by user so I can save them for next startup.
what is the best way to achieve that?

**Comment by Xyncgas** [comment]
Check your local time, round it to the nearest hour, seed a random number generator with that number then write a bunch files to a directory recording parameters for each window, then spawn a new window which will check for these files again and spawn a new window again, when there is no more files to read for spawning new window you just stop doing it, and return 0

**Reply by aliessmael** [reply]
Hi thanks, but my main issue is how to get and set position and size. there is no such parameter in maui window, also no OnPositionChanged or OnSizeChanged.

**Reply by Xyncgas** [reply]
Currently I've been using my own code that does a check at an interval for dimensions. I don't know if there is an api for window position.
There is https://gist.github.com/LanceMcCarthy/c8d23f5ba6157d5d53bee3b1a741f8b5
Meanwhile the BlazorWebView component inside MAUI also provides javascript api that does the same

**Comment by aliessmael** [comment, marked as answer]
Finally got it working, I will leave sample code for help

 ```csharp
public class MyWindow : Window    {
        public MyWindow()
        {
            Page = new MyPage();
            Title = "MyPage";
        }
#if WINDOWS
        protected override void OnCreated()
        {
            var window = this;
            IntPtr windowHandle  = WinRT.Interop.WindowNative.GetWindowHandle(window.Handler.PlatformView);
            WindowId windowId = Microsoft.UI.Win32Interop.GetWindowIdFromWindow(windowHandle);
            AppWindow appWindow = Microsoft.UI.Windowing.AppWindow.GetFromWindowId(windowId);

            RectInt32 rect = new();
            rect.X = 0;
            rect.Y = 0;
            rect.Width = 400;
            rect.Height = 400;

            appWindow.MoveAndResize( rect );
         
            appWindow.Changed += onWindowChanged;
        }

        void onWindowChanged( AppWindow appWindow , AppWindowChangedEventArgs args )
        {
            if( args.DidPositionChange )
            {
                Console.Write( appWindow.Position );
            }
            if( args.DidSizeChange )
            {
                Console.Write( appWindow.Size );
            }
        }
#endif  
    } 

```

**Reply by Xyncgas** [reply]
Looks like windows only code

**Reply by aliessmael** [reply]
Yes, because i am testing on windows, but same code should work on any PC platform I think, just required to add them to #if line.
